Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,363 @@
+  rw [← map_comp]; rfl
+
+variable {V : SSet}
+
+open SSet
+
+namespace Truncated
+
+local notation (priority := high) "[" n "]" => SimplexCategory.mk n
+
+/-- The map that picks up the initial vertex of a 2-simplex, as a morphism in the 2-truncated
+simplex category. -/
+def ι0₂ : [0]₂ ⟶ [2]₂ := δ₂ (n := 0) 1 ≫ δ₂ (n := 1) 1
+
+/-- The map that picks up the middle vertex of a 2-simplex, as a morphism in the 2-truncated
+simplex category. -/
+def ι1₂ : [0]₂ ⟶ [2]₂ := δ₂ (n := 0) 0 ≫ δ₂ (n := 1) 2
+
+/-- The map that picks up the final vertex of a 2-simplex, as a morphism in the 2-truncated
+simplex category. -/
+def ι2₂ : [0]₂ ⟶ [2]₂ := δ₂ (n := 0) 0 ≫ δ₂ (n := 1) 1
+
+/-- The initial vertex of a 2-simplex in a 2-truncated simplicial set. -/
+def ev0₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : OneTruncation₂ V := V.map ι0₂.op φ
+
+/-- The middle vertex of a 2-simplex in a 2-truncated simplicial set. -/
+def ev1₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : OneTruncation₂ V := V.map ι1₂.op φ
+
+/-- The final vertex of a 2-simplex in a 2-truncated simplicial set. -/
+def ev2₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : OneTruncation₂ V := V.map ι2₂.op φ
+
+/-- The 0th face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ0₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 0
+
+/-- The 1st face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ1₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 1
+
+/-- The 2nd face of a 2-simplex, as a morphism in the 2-truncated simplex category. -/
+def δ2₂ : [1]₂ ⟶ [2]₂ := δ₂ (n := 1) 2
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+0th face of a 2-simplex. -/
+def ev12₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev1₂ φ ⟶ ev2₂ φ :=
+  ⟨V.map δ0₂.op φ,
+    map_map_of_eq V (SimplexCategory.δ_comp_δ (i := 0) (j := 1) (by decide)).symm,
+    map_map_of_eq V rfl⟩
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+1st face of a 2-simplex. -/
+def ev02₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev0₂ φ ⟶ ev2₂ φ :=
+  ⟨V.map δ1₂.op φ, map_map_of_eq V rfl, map_map_of_eq V rfl⟩
+
+/-- The arrow in the ReflQuiver `OneTruncation₂ V` of a 2-truncated simplicial set arising from the
+2nd face of a 2-simplex. -/
+def ev01₂ {V : SSet.Truncated 2} (φ : V _[2]₂) : ev0₂ φ ⟶ ev1₂ φ :=
+  ⟨V.map δ2₂.op φ, map_map_of_eq V (SimplexCategory.δ_comp_δ (j := 1) le_rfl), map_map_of_eq V rfl⟩
+
+end Truncated
+open Truncated
+
+/-- The 2-simplices in a 2-truncated simplicial set `V` generate a hom relation on the free
+category on the underlying refl quiver of `V`. -/
+inductive HoRel₂ {V : SSet.Truncated 2} :
+    (X Y : Cat.FreeRefl (OneTruncation₂ V)) → (f g : X ⟶ Y) → Prop
+  | mk (φ : V _[2]₂) :
+    HoRel₂ _ _
+      (Quot.mk _ (Quiver.Hom.toPath (ev02₂ φ)))
+      (Quot.mk _ ((Quiver.Hom.toPath (ev01₂ φ)).comp
+        (Quiver.Hom.toPath (ev12₂ φ))))
+
+theorem HoRel₂.ext_triangle {V} (X X' Y Y' Z Z' : OneTruncation₂ V)
```

Comment:
Instead of this lemma, I would think that the following may be more useful:
```lean
theorem HoRel₂.mk' {V : SSet.Truncated 2} (φ : V _[2]₂) {X₀ X₁ X₂ : OneTruncation₂ V}
    (f₀₁ : X₀ ⟶ X₁) (f₁₂ : X₁ ⟶ X₂) (f₀₂ : X₀ ⟶ X₂)
    (h₀₁ : f₀₁.edge = V.map (δ₂ 2).op φ) (h₁₂ : f₁₂.edge = V.map (δ₂ 0).op φ)
    (h₀₂ : f₀₂.edge = V.map (δ₂ 1).op φ) :
    HoRel₂ _ _ (Quot.mk _ (Quiver.Hom.toPath f₀₂))
      (Quot.mk _ ((Quiver.Hom.toPath f₀₁).comp (Quiver.Hom.toPath f₁₂))) := by
  obtain rfl : X₀ = ev0₂ φ := by
    rw [← f₀₁.src_eq, h₀₁, ← FunctorToTypes.map_comp_apply, ← op_comp]
    -- a change in the definition of `ι0₂` may facilitate this proof...
    dsimp [ev0₂, ι0₂]
    apply congr_fun
    rw [← op_comp]
    congr 2
    ext (i : Fin 1)
    fin_cases i
    rfl
  obtain rfl : X₁ = ev1₂ φ := by
    rw [← f₀₁.tgt_eq, h₀₁, ← FunctorToTypes.map_comp_apply, ← op_comp]
    rfl
  obtain rfl : X₂ = ev2₂ φ := by
    rw [← f₁₂.tgt_eq, h₁₂, ← FunctorToTypes.map_comp_apply, ← op_comp]
    rfl
  obtain rfl : f₀₁ = ev01₂ φ := by ext; assumption
  obtain rfl : f₁₂ = ev12₂ φ := by ext; assumption
  obtain rfl : f₀₂ = ev02₂ φ := by ext; assumption
  constructor
```